src/NServiceBus.Core.Tests/Serializers/MessageDeserializerResolverTests.cs

```diff
@@ -0,0 +1,46 @@
+﻿namespace NServiceBus.Core.Tests.Serializers
+{
+    using System;
+    using NServiceBus.MessageInterfaces.MessageMapper.Reflection;
+    using NServiceBus.Serializers;
+    using NServiceBus.Serializers.Binary;
+    using NServiceBus.Serializers.Json;
+    using NServiceBus.Serializers.XML;
+    using NUnit.Framework;
+    using Conventions = NServiceBus.Conventions;
+
+    [TestFixture]
+    public class MessageDeserializerResolverTests
+    {
+        MessageDeserializerResolver resolver = null;
+
+        [SetUp]
+        public void Setup()
+        {
+            var mapper = new MessageMapper();
+            var xml = new XmlMessageSerializer(mapper, new Conventions());
+            var json = new JsonMessageSerializer(mapper);
+            var bson = new BsonMessageSerializer(mapper);
+            var binary = new BinaryMessageSerializer();
+
+            //resolver = new MessageDeserializerResolver(xml, new IMessageSerializer[] { json, bson, binary });
+        }
+
+        [TestCase(ContentTypes.Xml, typeof(XmlMessageSerializer))]
+        [TestCase(ContentTypes.Json, typeof(JsonMessageSerializer))]
+        [TestCase(ContentTypes.Bson, typeof(BsonMessageSerializer))]
+        [TestCase(ContentTypes.Binary, typeof(BinaryMessageSerializer))]
+        public void RetrievesSerializerByContentType(string contentType, Type expected)
+        {
+            var serializer = resolver.Resolve(contentType);
+            Assert.IsInstanceOf(expected, serializer);
+        }
+
+        [Test]
+        public void UnknownContentTypeFallsBackToXmlSerialization()
+        {
+            var serializer = resolver.Resolve("unknown/unsupported");
+            Assert.IsInstanceOf<XmlMessageSerializer>(serializer);
```

Comment:
so the desired behavior for an unmapped content type is to get some unhelpful serialization error from the xml serializer? ie if the content type if json and we try to deserialize it with xml. perhaps better to throw?